pmoves/services/hi-rag-gateway-v2/app.py

```
@@ -1612,7 +1723,10 @@ def geometry_decode_text(body: Dict[str, Any], _=Depends(require_tailscale)):
         raise HTTPException(501, "text decoder disabled")
     mode = (body.get("mode") or "geometry").lower()
     const_id = body.get("constellation_id")
-    k = int(body.get("k", 5))
+    try:
+        k = int(body.get("k", 5))
+    except (TypeError, ValueError):
+        raise HTTPException(400, "k must be an integer")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)